src/riak_kv_qry_buffers.erl

```diff
@@ -185,13 +219,10 @@ offset_to_scalar([A]) when is_integer(A), A >= 0 -> A.
           %% total chunks needed (== number of subqueries)
           chunks_need :: integer(),
 
-          %% flipped to true when chunks_need == chunks_got
-          all_chunks_received = false :: boolean(),
-
           %% total records stored (when complete)
           total_records = 0 :: non_neg_integer(),
 
-          %% %% iterator cache (need a working iterator support in eleveldb)
+          %% TODO: iterator cache
```

Comment:
Not sure what you think an "iterator cache" will be, but it could be a horrible thing ... leaving you in a position of never finding data.  Iterators lock in a "snapshot" of the database.  You likely want a fresh snapshot AFTER the query data finishes posting.